AdyenComponents/Apple Pay/ApplePayComponentExtensions.swift

```diff
@@ -34,14 +34,73 @@ extension ApplePayComponent: PKPaymentAuthorizationViewControllerDelegate {
         let details = ApplePayDetails(paymentMethod: applePayPaymentMethod,
                                       token: token,
                                       network: network,
-                                      billingContact: billingContact,
-                                      shippingContact: shippingContact)
+                                      billingContact: payment.billingContact,
+                                      shippingContact: payment.shippingContact,
+                                      shippingMethod: payment.shippingMethod)
         
         submit(data: PaymentComponentData(paymentMethodDetails: details, amount: self.amountToPay, order: order))
     }
+
+    public func paymentAuthorizationViewController(
+        _ controller: PKPaymentAuthorizationViewController,
+        didSelectShippingContact contact: PKContact,
+        handler completion: @escaping (PKPaymentRequestShippingContactUpdate) -> Void
+    ) {
+        guard let applePayDelegate = applePayDelegate else {
+            return completion(.init(paymentSummaryItems: applePayPayment.summaryItems))
+        }
+
+        let applePayPayment = applePayPayment
+        applePayDelegate.didUpdate(contact: contact,
+                                   for: applePayPayment,
+                                   completion: handlePaymentRequestUpdate(applePayPayment, completion))
+    }
+
+    public func paymentAuthorizationViewController(_ controller: PKPaymentAuthorizationViewController,
+                                                   didSelect shippingMethod: PKShippingMethod,
+                                                   handler completion: @escaping (PKPaymentRequestShippingMethodUpdate) -> Void) {
+        guard let applePayDelegate = applePayDelegate else {
+            return completion(.init(paymentSummaryItems: applePayPayment.summaryItems))
+        }
+
+        let applePayPayment = applePayPayment
+        applePayDelegate.didUpdate(shippingMethod: shippingMethod,
+                                   for: applePayPayment,
+                                   completion: handlePaymentRequestUpdate(applePayPayment, completion))
+    }
+
+    @available(iOS 15.0, *)
+    public func paymentAuthorizationViewController(_ controller: PKPaymentAuthorizationViewController,
+                                                   didChangeCouponCode couponCode: String,
+                                                   handler completion: @escaping (PKPaymentRequestCouponCodeUpdate) -> Void) {
+        guard let applePayDelegate = applePayDelegate else {
+            return completion(.init(paymentSummaryItems: applePayPayment.summaryItems))
+        }
+
+        let applePayPayment = applePayPayment
+        applePayDelegate.didUpdate(couponCode: couponCode,
+                                   for: applePayPayment,
+                                   completion: handlePaymentRequestUpdate(applePayPayment, completion))
```

Comment:
nice solution!
I think we need to call the `handlePaymentRequestUpdate` weakly to prevent any strong captures.